### ISSUE TITLE
Assets not loaded on wp-activate.php

### ISSUE THREAD

**Issue by websupporter** [inpsyde/assets#34]
**Describe the bug**
Lets say you have a multisite. The theme registers its main stylesheet like

```
new Style('handle', $url, Asset::FRONTEND);
```

The style will not be loaded on wp-activate.php. The problem is that wp-activate.php defines `WP_INSTALLING`. 

https://github.com/inpsyde/assets/blob/0f6337f892296a2bac57c855c8a62f42972d0368/src/Util/AssetHookResolver.php#L43

The AssetHookResolver relies on WpContext to define, whether we are in the front end, but for WpContext wp-activate.php is not considered to be in the front end because `WP_INSTALLING` is `true`:
https://github.com/inpsyde/wp-context/blob/main/src/WpContext.php#L65-L71

Not sure what the best way to resolve this would be. But at the moment, I think this is the most reasonable solution:

```
$isFront = $this->context->isFrontoffice() || $this->context->isInstalling()
```

**Comment by gmazzap** [comment]
The problem I see with the proposed solution is that `$this->context->isInstalling()` is not just `wp-activate.php`, but much more.

To solve the issue we IMO should:

1. Add a constant `Asset::ACTIVATE = 256`
2. Add a constant `Asset::THEME = 258` (which equals `Asset::ACTIVATE | Asset::FRONTEND`)
3. Add a constant `Asset::HOOK_ACTIVATE = 'activate_wp_head'`
4. In the [`Asset::HOOK_TO_LOCATION`](https://github.com/inpsyde/assets/blob/26afba74fdb9c71da39843643bb1735bddcd607a/src/Asset.php#L36) array, add `Asset::HOOK_ACTIVATE => Asset::ACTIVATE`
5. in the [`AssetHookResolver::resolve()`](https://github.com/inpsyde/assets/blob/26afba74fdb9c71da39843643bb1735bddcd607a/src/Util/AssetHookResolver.php#L44)  add something like:
```php
if ($this->context->isInstalling()) {
  return [Asset::HOOK_ACTIVATE];
}
```
6. in the [`AssetHookResolver::lastHook`](https://github.com/inpsyde/assets/blob/master/src/Util/AssetHookResolver.php#L82) add something like:
```php
if ($this->context->isInstalling()) {
  return Asset::HOOK_ACTIVATE;
}
```

That should already do it. When developers would need to add an assets that is available to the "usual frontend" + `wp-activate.php` they should use `new Style('handle', $url, Asset::THEME)`.

On top of that, using `Asset::ACTIVATE` would be possible to target _only_ the `wp-activate.php` page.

Additionally, we could make `Asset::THEME` the new default here: https://github.com/inpsyde/assets/blob/master/src/BaseAsset.php#L110, considering that `Asset::THEME` _includes_  `Asset::FRONTEND` that will not be a breaking change.

**Comment by Chrico** [comment]
Sounds good, but i'm not so sure if 2. will cover all (we e.G. also have embeds which are "Theme" as well, but currently not covered/tested). The 3. might contains not the correct hook. We can/should use the same hook as `Asset::FRONTEND` here.

The `Asset::FRONTEND` uses `wp_enqueue_scripts` hook which is triggered by `wp_enqueue_scripts()`-function in `wp-includes/script-loader.php`. The function is registered as callback to `wp_head` in `wp-includes/default-filters.php`. 

In `wp-activate.php` this is loaded via..

[`wp-activate.php`](https://github.com/WordPress/WordPress/blob/5.8/wp-activate.php#L12) 
-> [`wp-load.php`](https://github.com/WordPress/WordPress/blob/5.8/wp-load.php#L50) 
--> [`wp-config.php`](https://github.com/WordPress/WordPress/blob/5.8/wp-config-sample.php#L96) 
----> [`wp-settings.php`](https://github.com/WordPress/WordPress/blob/5.8/wp-settings.php#L134)
-----> [`wp-includes/default.filters.php`](https://github.com/WordPress/WordPress/blob/5.8/wp-includes/default-filters.php#L314)

The `wp-activate.php` uses [`get_header( 'wp-activate' );`](https://github.com/WordPress/WordPress/blob/5.8/wp-activate.php#L120) which falls back to `wp-includes/theme-compat/header.php` which calls [`wp_head()`](https://github.com/WordPress/WordPress/blob/5.8/wp-includes/theme-compat/header.php#L49).

So in short: On 3. we might can/should use `Asset::HOOK_ACTIVATE = 'wp_enqueue_scripts'`. 🙈

**Comment by gmazzap** [comment]
> We also have embeds which are "Theme" as well

`EMBED` is [a type of query](https://github.com/WordPress/WordPress/blob/5.8/wp-includes/query.php#L885). We don't have things like `Assets::SINGULAR` or `Assets::ARCHIVE` or `Assets::SEARCH`... because those are all types of queries, and all belongs to `FRONTEND`.

---

> we might can/should use Asset::HOOK_ACTIVATE = 'wp_enqueue_scripts'

In `wp-activate.php` WordPress adds [`do_activate_header()` to `wp_head` hook](https://github.com/WordPress/WordPress/blob/5.8/wp-activate.php#L98) and `do_activate_header()` does nothing more than [`do_action('activate_wp_head')`](https://github.com/WordPress/WordPress/blob/5.8/wp-activate.php#L96) 

That means, that when visiting `wp-activate.php` what WordPress does can be summarized in:

```
add_action('wp_head', fn() => do_action('wp_enqueue_scripts'), 1);
add_action('wp_head', fn() => do_action('activate_wp_head'), 10);
```

So by using `activate_wp_head` and `wp_enqueue_scripts` there isn't much difference... `activate_wp_head` is called slightly later (priority 10 instead of priority 1), but `activate_wp_head` is *only* called on `wp-activate.php`, opposed to `wp_enqueue_scripts` which is called in many other situations.

If we map `Assets::HOOK_ACTIVATE` to `'wp_enqueue_scripts'` assets added with `Asset::ACTIVATE` location would be added in every frontend request and not _only_ on WP activate.

**Comment by Chrico** [comment]
> So by using activate_wp_head and wp_enqueue_scripts there isn't much difference... activate_wp_head is called slightly later (priority 10 instead of priority 1), but activate_wp_head is only called on wp-activate.php, opposed to wp_enqueue_scripts which is called in many other situations.

Hmm thats true. So we're going to use `activate_wp_head` and use `inpsyde/wp-context` with min v1.3 for that. I created a pull here: https://github.com/inpsyde/assets/pull/35  - @websupporter feel free to test :)

**Comment by Chrico** [comment]
This is part of the newest 2.7 release - see here: https://github.com/inpsyde/assets/releases/tag/2.7

@websupporter feel free to close when this issue is solved for you. :)

**Comment by websupporter** [comment]
Thanks @Chrico @gmazzap
